rll/upgrade.sh

```diff
@@ -89,36 +78,36 @@ fi
 re="^\s*${current_version}\s*:\s*(\S+)\s*$"
 match=`curl --silent --show-error --retry 3 ${UPGRADES_FILE_LOCATION} | egrep ${re} || true`
 if [[ "$match" =~ $re ]]; then
-  desired=${BASH_REMATCH[1]}
+  desired="${BASH_REMATCH[1]}"
 else
   echo "Cannot determine latest version from upgrade file"
   echo "Tried to match /^${current_version}:/ in ${UPGRADES_FILE_LOCATION}"
   exit 0
 fi
 
-if [[ $desired == $current_version ]]; then
-  echo "RightLink is already up-to-date (current=$current_version)"
+if [[ "$desired" == "$current_version" ]]; then
+  echo "RightLink is already up-to-date (current=${current_version})"
   exit 0
 fi
 
 echo "RightLink needs update:"
-echo "  from current=$current_version"
-echo "  to   desired=$desired"
+echo "  from current=${current_version}"
+echo "  to   desired=${desired}"
 
-echo "downloading RightLink version '$desired'"
+echo "downloading RightLink version '${desired}'"
 
 # Download new version
 cd /tmp
-rm -rf rightlink rightlink.tgz
-curl --silent --show-error --retry 3 --output rightlink.tgz $prefix_url/$desired/rightlink.tgz
+sudo rm -rf rightlink rightlink.tgz
+sudo curl --silent --show-error --retry 3 --output rightlink.tgz https://rightlink.rightscale.com/rll/${desired}/rightlink.tgz
 tar zxf rightlink.tgz || (cat rightlink.tgz; exit 1)
 
 # Check downloaded version
-mv rightlink/rightlink ${rl_bin}-new
+sudo mv rightlink/rightlink ${rl_bin}-new
```

Comment:
Do not need sudo here since we donwloaded the tar ball and expanded it as the rightlink user.